src/nix-env/nix-env.cc

```diff
@@ -564,17 +577,23 @@ static void installDerivations(Globals & globals,
 
 static void opInstall(Globals & globals, Strings opFlags, Strings opArgs)
 {
+    std::string priority = "";
```

Comment:
```suggestion
    std::optional<int> priority;
```
We want to validate that the argument is an integer. Currently I think you can pass anything to `--profile`.

Reply:
Originally i had done that, but then I looked at the impl of `opSetFlag` and it also stores priority as a string (and does not validate it). The string gets validated to be an int somewhere closer to the point of use.

Happy to validate up front as well.